### PR TITLE
chore: create verification-link-expired-modal

### PR DESCRIPTION
**Pull request by suisin-deriv**
## Changes:

Added Verification Link Expired Modal

### Screenshots:

Screenshot 2024-05-17 at 2 31 46 PM

### CODE REVIEW

File: packages/account/src/Sections/Profile/PhoneVerification/__test__/verification-link-expired-modal.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            <StoreProvider store={mock_store}>
+                <VerificationLinkExpiredModal
+                    should_show_verification_link_expired_modal
+                    setShouldShowVerificationLinkExpiredModal={mockSetShowVerificationLinkExpiredModal}
+                />
+            </StoreProvider>
+        );
+    };
+
+    it('it should render VerificationLinkExpiredModal', () => {
```
Comment (hasan-deriv) [inline packages/account/src/Sections/Profile/PhoneVerification/__test__/verification-link-expired-modal.spec.tsx:49]:
```suggestion
    it('should render VerificationLinkExpiredModal', () => {
```

Comment (hasan-deriv) [reply packages/account/src/Sections/Profile/PhoneVerification/__test__/verification-link-expired-modal.spec.tsx:49]: It should be like this in all other tests as well..

**Review by hasan-deriv — CHANGES_REQUESTED** [review]

File: packages/account/src/Sections/Profile/PhoneVerification/__test__/verification-link-expired-modal.spec.tsx (hunk 1 above)
Comment (suisin-deriv) [reply packages/account/src/Sections/Profile/PhoneVerification/__test__/verification-link-expired-modal.spec.tsx:49]: oops will change it 😁

File: packages/account/src/Sections/Profile/PhoneVerification/verification-link-expired-modal.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        should_show_verification_link_expired_modal,
+        setShouldShowVerificationLinkExpiredModal,
+    }: TVerificationLinkExpiredModal) => {
+        const history = useHistory();
+        const handleCancelButton = () => {
+            setShouldShowVerificationLinkExpiredModal(false);
+            history.goBack();
+        };
+        const { ui } = useStore();
+        const { is_mobile } = ui;
```
Comment (utkarsha-deriv) [inline packages/account/src/Sections/Profile/PhoneVerification/verification-link-expired-modal.tsx:24]: Should we use `isMobile` from `useDevice` hook?

Comment (suisin-deriv) [reply packages/account/src/Sections/Profile/PhoneVerification/verification-link-expired-modal.tsx:24]: Tried to implement the useDevice here but it's giving me error: 
Screenshot 2024-05-17 at 3 20 28 PM
last time i had this error when implementing the quill-ui, maybe need to ask the deriv-com/ui team to help solve this first

File: packages/account/src/Sections/Profile/PhoneVerification/verification-link-expired-modal.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+        return (
+            <Modal
+                isMobile={is_mobile}
+                showHandleBar
+                isOpened={should_show_verification_link_expired_modal}
+                primaryButtonCallback={() => setShouldShowVerificationLinkExpiredModal(false)}
```
Comment (utkarsha-deriv) [inline packages/account/src/Sections/Profile/PhoneVerification/verification-link-expired-modal.tsx:31]: Should we also create a separate function for this callback as we will add some api call for sending new link? 🙏

**Review by hasan-deriv — APPROVED** [review]

File: packages/account/src/Sections/Profile/PhoneVerification/verification-link-expired-modal.tsx (hunk 3 above)
Comment (suisin-deriv) [reply packages/account/src/Sections/Profile/PhoneVerification/verification-link-expired-modal.tsx:31]: yeap but the API is not ready for now, will implement that when the API is ready. For now we will temporary use useState to mock closing the modal

File: packages/account/src/Sections/Profile/PhoneVerification/verification-link-expired-modal.tsx (hunk 2 above)
Comment (utkarsha-deriv) [reply packages/account/src/Sections/Profile/PhoneVerification/verification-link-expired-modal.tsx:24]: yes, let's ask them 🙏

**Review by utkarsha-deriv — APPROVED** [review]

**Review by yauheni-deriv — APPROVED** [review]

**Review by amina-deriv — APPROVED** [review]

**Pull request merged**